src/Simplex/Chat/Messages.hs

```diff
@@ -537,6 +538,7 @@ deriving instance Show (CIContent d)
 data RcvGroupEvent
   = RGEMemberAdded {groupMemberId :: GroupMemberId, profile :: Profile} -- CRJoinedGroupMemberConnecting
   | RGEMemberConnected -- CRUserJoinedGroup, CRJoinedGroupMember, CRConnectedToGroupMember
+  | RGEMemberConnectedIncognito {mainProfile :: Profile, incognitoProfile :: Profile} -- CRUserJoinedGroup, CRJoinedGroupMember, CRConnectedToGroupMember
```

Comment:
not sure I understand why you need both profiles here, at most you should need one (if any) and it can be merged with RGEMemberConnected with Maybe field

Reply:
... because one of the two profiles is the sender in the received message

Reply:
... or maybe both, if the second profile is the one belonging to the contact